sql/src/main/java/io/crate/operation/user/User.java

```diff
@@ -57,14 +58,28 @@ public String name() {
         return privileges;
     }
 
+    public boolean isSuperUser() {
+        return roles.contains(Role.SUPERUSER);
+    }
+
+    public boolean hasPrivilege(Privilege.Type type, Privilege.Clazz clazz, Privilege.State state) {
+        Optional<Privilege> hasPrivilege = privileges.stream()
```

Comment:
thinks its much more convenient to create a Privilege and pass it in here, so a privileges.contains() check is sufficient